src_plugins/show/ppx_deriving_show.cppo.ml

```diff
@@ -317,14 +312,56 @@ let str_of_type ~options ~path ({ ptype_loc = loc } as type_decl) =
          (Ppx_deriving.sanitize ~quoter (polymorphize prettyprinter));
    Vb.mk ~attrs:[no_warn_32] (Pat.constraint_ show_var show_type) (polymorphize stringprinter);]
 
-let () =
-  let loc = !Ast_helper.default_loc in
-  Ppx_deriving.(register (create deriver
-    ~core_type: (Ppx_deriving.with_quoter (fun quoter typ ->
-      [%expr fun x -> Ppx_deriving_runtime.Format.asprintf "%a" (fun fmt -> [%e expr_of_typ quoter typ]) x]))
-    ~type_decl_str: (fun ~options ~path type_decls ->
-      [Str.value Recursive (List.concat (List.map (str_of_type ~options ~path) type_decls))])
-    ~type_decl_sig: (fun ~options ~path type_decls ->
-      List.concat (List.map (sig_of_type ~options ~path) type_decls))
-    ()
-  ))
+(* TODO: add to ppxlib? *)
+let ebool: _ Ast_pattern.t -> _ Ast_pattern.t =
+  Ast_pattern.map1' ~f:(fun loc -> function
+    | [%expr true] -> true
+    | [%expr false] -> false
+    | _ -> Location.raise_errorf ~loc "with_path should be a boolean")
+let args () = Deriving.Args.(empty +> arg "with_path" (ebool __))
```

Comment:
Is there a reason you add a unit argument here?

Reply:
In some ppxlib example, it is used to have the toplevel display a type which is not affected by the value retriction. So maybe that comes from one such example copied and copied... In any case, I don't think it's useful here!